pkg/controller/core/workload_controller.go

```diff
@@ -801,70 +874,90 @@ func (h *resourceUpdatesHandler) queueReconcileForPending(ctx context.Context, _
 	}
 }
 
-type workloadCqHandler struct {
-	client client.Client
+type workloadQueueHandler struct {
+	r *WorkloadReconciler
 }
 
-var _ handler.EventHandler = (*workloadCqHandler)(nil)
+var _ handler.EventHandler = (*workloadQueueHandler)(nil)
 
 // Create is called in response to a create event.
-func (w *workloadCqHandler) Create(ctx context.Context, ev event.CreateEvent, wq workqueue.RateLimitingInterface) {
-	if cq, isQueue := ev.Object.(*kueue.ClusterQueue); isQueue {
-		w.queueReconcileForWorkloads(ctx, cq.Name, wq)
+func (w *workloadQueueHandler) Create(ctx context.Context, ev event.CreateEvent, wq workqueue.RateLimitingInterface) {
+	if cq, isCq := ev.Object.(*kueue.ClusterQueue); isCq {
+		w.queueReconcileForWorkloadsOfClusterQueue(ctx, cq.Name, wq)
+	}
+	if lq, isLq := ev.Object.(*kueue.LocalQueue); isLq {
+		log := ctrl.LoggerFrom(ctx).WithValues("localQueue", klog.KObj(lq))
+		ctx = ctrl.LoggerInto(ctx, log)
+		w.queueReconcileForWorkloadsOfLocalQueue(ctx, lq, wq)
 	}
 }
 
 // Update is called in response to an update event.
-func (w *workloadCqHandler) Update(ctx context.Context, ev event.UpdateEvent, wq workqueue.RateLimitingInterface) {
-	log := ctrl.LoggerFrom(ctx).WithValues("clusterQueue", klog.KObj(ev.ObjectNew))
-	ctx = ctrl.LoggerInto(ctx, log)
-	log.V(5).Info("Workload cluster queue update event")
-	oldCq, oldIsQueue := ev.ObjectOld.(*kueue.ClusterQueue)
-	newCq, newIsQueue := ev.ObjectNew.(*kueue.ClusterQueue)
+func (w *workloadQueueHandler) Update(ctx context.Context, ev event.UpdateEvent, wq workqueue.RateLimitingInterface) {
+	oldCq, oldIsCq := ev.ObjectOld.(*kueue.ClusterQueue)
+	newCq, newIsCq := ev.ObjectNew.(*kueue.ClusterQueue)
+	if oldIsCq && newIsCq {
+		log := ctrl.LoggerFrom(ctx).WithValues("clusterQueue", klog.KObj(ev.ObjectNew))
+		ctx = ctrl.LoggerInto(ctx, log)
+		log.V(5).Info("Workload cluster queue update event")
 
-	if !oldIsQueue || !newIsQueue {
-		return
+		if !newCq.DeletionTimestamp.IsZero() ||
+			!utilslices.CmpNoOrder(oldCq.Spec.AdmissionChecks, newCq.Spec.AdmissionChecks) ||
+			!gocmp.Equal(oldCq.Spec.AdmissionChecksStrategy, newCq.Spec.AdmissionChecksStrategy) ||
+			!ptr.Equal(oldCq.Spec.StopPolicy, newCq.Spec.StopPolicy) {
+			w.queueReconcileForWorkloadsOfClusterQueue(ctx, newCq.Name, wq)
+		}
```

Comment:
```suggestion
			w.queueReconcileForWorkloadsOfClusterQueue(ctx, newCq.Name, wq)
		}
		return
```
ther is no point to check if it's a lq if it is a cq

Reply:
Done